### DISCUSSION TITLE
End quarto preview with serving address

### DISCUSSION THREAD

**Discussion by gshotwell** [quarto-dev/quarto-cli#6890] [Feature Requests]
### Description

One of the great things about the Sreamlit CLI is that while serving the site they always end their CLI messages with the address. This is great because if you close the tab you can get back to it easily. Right now in quarto the `Browse at http://localhost:1234/` comes at the beginning and can get buried under other messages as the user navigates or the site rebuilds in response to file changes. Would it be possible to append the browse message at the end of other messages?

**Comment by cscheid** [comment]
It's easy for us to "just implement it", but the problem we faced with this before is that there are a number of downstream tools that depend on the output generated by `quarto preview`

@jjallaire Do you know if we have code in RStudio that depends on this behavior or is it just the VSC extension?

**Comment by mcanouil** [comment]
I am pretty sure this was discussed previously in an Issue or Discussion (can't search for it now).

Edit: indeed a duplicate of:

- https://github.com/quarto-dev/quarto-cli/discussions/4779

**Reply by gshotwell** [reply]
Ah okay, sorry for missing that. I commented over there so we can probably close this.

**Comment by jjallaire** [comment]
Yes, there are likely tools that depend on the behavior we have now so changing things is delicate (and we are generally biased against)

**Comment by mcanouil** [comment]
Quick note, the port can be set which will avoid having to remember the preview URL, see <https://quarto.org/docs/reference/projects/options.html#preview>.
